cdc/owner/owner.go

```diff
@@ -410,21 +410,36 @@ func (o *ownerImpl) clusterVersionConsistent(captures map[model.CaptureID]*model
 }
 
 func (o *ownerImpl) handleDrainCaptures(query *scheduler.Query, done chan<- error) {
-	changefeedWithTableCount := 0
-	totalTableCount := 0
+	var (
+		changefeedWithTableCount int
+		totalTableCount          int
+		err                      error
+	)
 	for _, changefeed := range o.changefeeds {
-		count := changefeed.scheduler.DrainCapture(query.CaptureID)
+		count, e := changefeed.scheduler.DrainCapture(query.CaptureID)
```

Comment:
Please check error right after `DrainCapture`

```suggestion
		count, e := changefeed.scheduler.DrainCapture(query.CaptureID)
		if e != nil {
			err = e
			break
		}
```